File: services/dotnet-microservices/NotificationService/Domain/Models/Notification.cs
namespace NotificationService.Domain.Models;

public record Notification(
    Guid Id,
    Guid? UserId,
    string Type,
    string Channel,
    string Subject,
    string Body,
    string Status,
    DateTime CreatedAt,
    DateTime? SentAt
);

public record CreateNotificationRequest(
    Guid? UserId,
    string Type,
    string Channel,
    string Subject,
    string Body
);

public static class NotificationTypes
{
    public const string UserCreated = "user_created";
    public const string OrderConfirmed = "order_confirmed";
    public const string OrderCancelled = "order_cancelled";
    public const string PaymentProcessed = "payment_processed";
    public const string PaymentFailed = "payment_failed";
    public const string LowStock = "low_stock";
}

public static class NotificationChannels
{
    public const string Email = "email";
    public const string Sms = "sms";
    public const string Push = "push";
}

File: services/nodejs-microservices/notification-service/src/domain/models/Notification.ts
export interface Notification {
  id: string;
  type: NotificationType;
  recipient: string;
  subject: string;
  body: string;
  status: NotificationStatus;
  sentAt?: Date;
  createdAt: Date;
}

export enum NotificationType {
  ORDER_CONFIRMATION = 'ORDER_CONFIRMATION',
  ORDER_CANCELLED = 'ORDER_CANCELLED',
  PAYMENT_RECEIVED = 'PAYMENT_RECEIVED',
  PAYMENT_REFUNDED = 'PAYMENT_REFUNDED',
  WELCOME = 'WELCOME',
  LOW_STOCK = 'LOW_STOCK',
}

export enum NotificationStatus {
  PENDING = 'PENDING',
  SENT = 'SENT',
  FAILED = 'FAILED',
}
